apps/recnet-api/src/modules/article/article.service.ts

```diff
@@ -19,39 +20,68 @@ export class ArticleService {
     private readonly digitalLibraryService: DigitalLibraryService | null
   ) {}
 
+  // public async getArticleByLink(
+  //   link: string
+  // ): Promise<GetArticleByLinkResponse> {
+  //   let unifiedLink = link;
+  //
+  //   // If digital library service is available, try to get the unified link
+  //   if (this.digitalLibraryService) {
+  //     unifiedLink = await this.digitalLibraryService.getUnifiedLink(link);
+  //   }
+  //
+  //   // Try to find the article in the database using the unified link
+  //   let article = await this.articleRepository.findArticleByLink(unifiedLink);
+  //   if (article) {
+  //     return { article };
+  //   }
+  //
+  //   // If no article found, try to get metadata using the digital library service
+  //   if (this.digitalLibraryService) {
+  //     try {
+  //       const metadata = await this.digitalLibraryService.getMetadata(link);
+  //       const articleInput = this.transformMetadata(metadata, unifiedLink);
+  //
+  //       article = await this.articleRepository.createArticle(articleInput);
+  //       return { article };
+  //     } catch (error) {
+  //       this.logger.error(error);
+  //     }
+  //   }
+  //
+  //   // If no article found and no metadata available, return null
+  //   return {
+  //     article: null,
+  //   };
+  // }
+
   public async getArticleByLink(
-    link: string
+    link: string,
+    fetchFromDigitalLibrary: boolean = true
   ): Promise<GetArticleByLinkResponse> {
     let unifiedLink = link;
 
-    // If digital library service is available, try to get the unified link
-    if (this.digitalLibraryService) {
+    if (fetchFromDigitalLibrary && this.digitalLibraryService) {
       unifiedLink = await this.digitalLibraryService.getUnifiedLink(link);
     }
 
-    // Try to find the article in the database using the unified link
     let article = await this.articleRepository.findArticleByLink(unifiedLink);
     if (article) {
       return { article };
     }
 
-    // If no article found, try to get metadata using the digital library service
-    if (this.digitalLibraryService) {
+    if (fetchFromDigitalLibrary && this.digitalLibraryService) {
       try {
         const metadata = await this.digitalLibraryService.getMetadata(link);
         const articleInput = this.transformMetadata(metadata, unifiedLink);
-
         article = await this.articleRepository.createArticle(articleInput);
         return { article };
       } catch (error) {
         this.logger.error(error);
       }
     }
 
-    // If no article found and no metadata available, return null
-    return {
-      article: null,
-    };
+    return { article: null };
```

Comment:
Btw, I think leaving the comment there would be helpful for developers to understand. Since the original logic might not be easy to understand.